packages/dd-trace/src/plugins/util/git.js

```diff
@@ -36,8 +36,24 @@ function isShallowRepository () {
 }
 
 function unshallowRepository () {
-  sanitizedExec('git', ['config', 'remote.origin.partialclonefilter', '"blob:none"'])
-  sanitizedExec('git', ['fetch', '--shallow-since="1 month ago"', '--update-shallow', '--refetch'])
+  const gitVersionString = sanitizedExec('git', ['version'])
+  const gitVersionSplit = gitVersionString.split(' ')[2].split('.')
+  const gitVersionMain = parseInt(gitVersionSplit[0])
+  const gitVersionSecondary = parseInt(gitVersionSplit[1])
```

Comment:
let's extract this into a separate function `getGitVersion`
```javascript
function getGitVersion () {
  const gitVersionString = sanitizedExec('git', ['version'])
  const gitVersionMatches = gitVersionString.match(/git version (\d+)\.(\d+)\.(\d+)/)
  try {
    return {
      major: parseInt(gitVersionMatches[1]),
      minor: parseInt(gitVersionMatches[2]),
      patch: parseInt(gitVersionMatches[3])
    }  
  } catch (e) {
    return null
  }
}
```

and then we can do the following in this function:

```javascript
const gitVersion = getGitVersion()
if (!gitVersion) {
  log.warn('Git version could not be extracted, so git unshallow will not proceed')
  return
}
if (gitVersion.major < 2 || (gitVersion.major === 2 && gitVersion.minor < 27) {
  log.warn('Git version is <=2.27, so git unshallow will not proceed')
  return 
}
// the rest of what you have is fine 
```

Reply:
Great suggestion, will implement this.